app/services/analytics.rb

```diff
@@ -4,7 +4,54 @@ class Analytics
   include AnalyticsEvents
   prepend Idv::AnalyticsEventsEnhancer
 
-  attr_reader :user, :request, :sp, :session, :ahoy
+  # Analytics middleware that sends the event to Ahoy
+  class AhoyMiddleware
+    attr_reader :ahoy
+
+    def initialize(ahoy: nil, request: nil)
+      @ahoy = ahoy || Ahoy::Tracker.new(request:)
+    end
+
+    def call(event)
+      ahoy.track(event[:name], event[:properties])
+      nil
```

Comment:
My thinking is that middleware should be able to modify / augment the event by returning a new one. Return nil and the old one just gets used